src/postprocessor/compress.rs

```diff
@@ -92,42 +174,63 @@ impl Postprocessor for Lz4 {
     fn process(&mut self, _ingres_ns: u64, _egress_ns: u64, data: &[u8]) -> Result<Vec<Vec<u8>>> {
         use lz4::EncoderBuilder;
         let buffer = Vec::<u8>::new();
-        let mut encoder = EncoderBuilder::new().level(4).build(buffer)?;
+        let mut encoder = EncoderBuilder::new()
+            .level(self.compression_level)
+            .build(buffer)?;
         encoder.write_all(data)?;
         Ok(vec![encoder.finish().0])
     }
 }
+impl Default for Lz4 {
+    fn default() -> Self {
+        Self {
+            compression_level: 4,
+        }
+    }
+}
 
-#[derive(Clone, Default, Debug)]
-struct Zstd {}
+#[derive(Clone, Debug)]
+struct Zstd {
+    compression_level: i32,
+}
+impl Zstd {
+    pub fn with_config(level: i32) -> Self {
```

Comment:
Adding this here but it applies to all compressors. The constructor should probably valid the compression level to be correct, we don't want a runtime failure because someone defines level 999999 and only 20 is valid.